src/main/java/mate/academy/shop/jdbc/ItemDaoJdbcImpl.java

```diff
@@ -28,135 +28,150 @@ public ItemDaoJdbcImpl(Connection connection) {
-            return item;
+    public void delete(Long id) {
+        String query = "DELETE FROM items WHERE item_id=?;";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            statement.executeUpdate();
         } catch (SQLException e) {
-            logger.warn("Can't create the item with id=" + id);
-        } finally {
-            if (stmt != null) {
-                try {
-                    stmt.close();
-                } catch (SQLException e) {
-                    logger.warn("Can't close statement ", e);
-                }
-            }
+            logger.warn("Can't delete the item with id=" + id);
         }
-        return null;
     }
 
     @Override
     public List<Item> getAllItems() {
         List<Item> list = new ArrayList<>();
-        Statement stmt = null;
-        String query = String.format("SELECT * FROM %s.items;", DB_NAME);
-        try {
-            stmt = connection.createStatement();
-            ResultSet rs = stmt.executeQuery(query);
-            while (rs.next()) {
-                long itemId = rs.getLong(ITEM_ID_CONST);
-                String name = rs.getString(ITEM_NAME_CONST);
-                double price = rs.getDouble(ITEM_PRICE_CONST);
+        String query = "SELECT * FROM items;";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery(query);
+            while (resultSet.next()) {
+                long itemId = resultSet.getLong(ITEM_ID_CONST);
+                String name = resultSet.getString(ITEM_NAME_CONST);
+                double price = resultSet.getDouble(ITEM_PRICE_CONST);
                 Item item = new Item(itemId);
                 item.setName(name);
                 item.setPrice(price);
                 list.add(item);
             }
         } catch (SQLException e) {
             logger.warn("Can't get items");
-        } finally {
-            if (stmt != null) {
-                try {
-                    stmt.close();
-                } catch (SQLException e) {
-                    logger.warn("Can't close statement ", e);
-                }
+        }
+        return list;
+    }
+
+    public List<Item> getItemForBucket(Long bucketId) {
+        List<Item> list = new ArrayList<>();
+        String query = "SELECT * FROM items_buckets where bucket_id=?;";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, bucketId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Long itemId = resultSet.getLong(ITEM_ID_CONST);
+                list.add(get(itemId));
             }
+        } catch (SQLException e) {
+            logger.warn("Can't find items for bucket with id=" + bucketId);
         }
         return list;
     }
+
+    public void deleteAllItemsFromBucket(Long bucketId) {
```

Comment:
the same about this method and all other method where we use item and bucket

Reply:
done